metrics/system_tag.go

```diff
@@ -153,3 +153,91 @@ func (i *SystemTagSet) UnmarshalText(data []byte) error {
 	}
 	return nil
 }
+
+// NullSystemTagSet is a wrapper around SystemTagSet like guregu/null
+type NullSystemTagSet struct {
+	Set   *SystemTagSet
+	Valid bool
+}
+
+// ToNullSystemTagSet converts list of tags to NullSystemTagSet
+func ToNullSystemTagSet(tags []string) NullSystemTagSet {
+	return NullSystemTagSet{
+		Set:   ToSystemTagSet(tags),
+		Valid: true,
+	}
+}
+
+// NewNullSystemTagSet returns valid (Valid: true) SystemTagSet
+func NewNullSystemTagSet(tags ...SystemTag) NullSystemTagSet {
+	return NullSystemTagSet{
+		Set:   NewSystemTagSet(tags...),
+		Valid: true,
+	}
+}
+
+// Add adds a tag to tag set.
+func (n *NullSystemTagSet) Add(tag SystemTag) {
+	if n.Set == nil {
+		n.Set = new(SystemTagSet)
+		n.Valid = true
+	}
+	n.Set.Add(tag)
+}
+
+// Has checks a tag included in tag set.
+func (n NullSystemTagSet) Has(tag SystemTag) bool {
+	if !n.Valid {
+		return false
+	}
+	return n.Set.Has(tag)
+}
+
+// Map returns the EnabledTags with current value from SystemTagSet
+func (n NullSystemTagSet) Map() EnabledTags {
+	return n.Set.Map()
+}
+
+// SetString returns comma separated list of the string representation of all values in the set
+func (n NullSystemTagSet) SetString() string {
+	return n.Set.SetString()
+}
```

Comment:
I take it that without this additional methods the change is much bigger ?

Reply:
I might be missing something, but some of these methods are used widely (for instance, a quick search gives ~30 occurrences of `SystemTags.Has(...)`, so the only alternative I see is replacing each of them by: `SystemTags.Set.Has(...)`.

Do you have any other suggestion? 🤔

Reply:
Any news? @mstoykov 🙏🏻

Reply:
Sorry, I started writing and then apparently never published this.


Given how big this change is and that it now mixes Null types with the actual implementation of how it is used - which we never do, I am kind of for not going forward with this. 

I kind of prefer if we just use Null type and then fix it everywhere, but it might be *a bit* much. Arguably we will have to do this huge breaking change when we move to https://github.com/grafana/croconf or for any other fix for #883 . But not certain if we should do this before.

So at this point I kind of thing that your proposal to just patch `envconfig` might be better :thinking: 

Can you open a PR so we can at least look at it in parallel. 

cc @codebien I guess this also is your concern

Reply:
Here's the PR: https://github.com/mstoykov/envconfig/pull/1
cc/ @codebien